StreamChatClient_v3/WebSocketClient/Events/ChannelEvents.swift

```diff
@@ -6,29 +6,32 @@
 import Foundation
 
 public protocol ChannelEvent: Event {
-    associatedtype ExtraData: ExtraDataTypes
-    
-    var channelId: ChannelId { get }
+    var cid: ChannelId { get }
 }
 
 public struct AddedToChannel<ExtraData: ExtraDataTypes>: ChannelEvent {
     public static var eventRawType: String { "notification.added_to_channel" }
     
-    public var channelId: ChannelId {
-        channelPayload.channel.cid
-    }
-    
-    let channelPayload: ChannelPayload<ExtraData>
+    public let cid: ChannelId
+    let eventPayload: AnyEventPayload
     
-    init?(from eventResponse: EventResponse<ExtraData>) throws {
-        guard eventResponse.eventType == Self.eventRawType else { return nil }
-        guard let channel = eventResponse.channelPayload else {
-            throw ClientError.EventDecodingError("`channel` field can't be `nil` for the RemovedFromChannel event.")
+    init?(from eventPayload: EventPayload<ExtraData>) throws {
+        guard eventPayload.eventType == Self.eventRawType else { return nil }
+        guard eventPayload.channelPayload != nil else {
+            throw ClientError.EventDecodingError("`channel` field can't be `nil` for the `AddedToChannel` event.")
         }
-        self.init(channelPayload: channel)
+        
+        guard eventPayload.cid != nil else {
+            throw ClientError.EventDecodingError("`cid` field can't be `nil` for the `AddedToChannel` event.")
+        }
+        
+        self.init(cid: try! ChannelId(cid: eventPayload.cid!), eventPayload: eventPayload)
```

Comment:
Looks a little weird with `cid` flipping. 
Why you can't make easier like this:
```suggestion
        self.init(cid: eventPayload.cid, eventPayload: eventPayload)
```
`cid` presents for a subset of events and we use structs. Can you define a protocol like `EventPayloadWithChannelId` or  `ChannelRelatedEventPayload`.

Reply:
Oh, this is a mistake, `EventPayload.cid` should be `ChannelId` not `String`. Thanks

Reply:
Fixed in 327f11227f1090c50638a6c2b33ea1f4c7f7abb1 4d8bb460a5498e08a6a4c7a84294c49b04c9f0cf ea7a51e6d96a9b87c1daef15440d353bee26aa79 546c62735b762e10a85bc40d5cf955923153958a